$MirOS: ports/security/gnupg/patches/patch-util_miscutil_c,v 1.2 2007/03/11 20:41:16 tg Exp $
--- util/miscutil.c.orig	Tue Oct 23 07:55:08 2007
+++ util/miscutil.c	Tue Apr  1 15:22:24 2008
@@ -123,8 +123,8 @@ strtimestamp( u32 stamp )
     }
     else {
         tp = gmtime( &atime );
-        sprintf(buffer,"%04d-%02d-%02d",
-                1900+tp->tm_year, tp->tm_mon+1, tp->tm_mday );
+        sprintf(buffer,"%04ld-%02d-%02d",
+                1900L+(long)(tp->tm_year), tp->tm_mon+1, tp->tm_mday );
     }
     return buffer;
 }
@@ -145,8 +145,8 @@ isotimestamp (u32 stamp)
     }
     else {
         tp = gmtime( &atime );
-        sprintf(buffer,"%04d-%02d-%02d %02d:%02d:%02d",
-                1900+tp->tm_year, tp->tm_mon+1, tp->tm_mday,
+        sprintf(buffer,"%04ld-%02d-%02d %02d:%02d:%02d",
+                1900L+(long)(tp->tm_year), tp->tm_mon+1, tp->tm_mday,
                 tp->tm_hour, tp->tm_min, tp->tm_sec);
     }
     return buffer;
